Crash report 3.9-rc-1: ClassCastException in SettingsActivity

```
java.lang.RuntimeException: Unable to start activity ComponentInfo{org.wordpress.android/org.wordpress.android.ui.prefs.SettingsActivity}: java.lang.ClassCastException: android.preference.PreferenceCategory cannot be cast to android.preference.PreferenceScreen
       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2325)
       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2387)
       at android.app.ActivityThread.access$800(ActivityThread.java:151)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1303)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:135)
       at android.app.ActivityThread.main(ActivityThread.java:5254)
       at java.lang.reflect.Method.invoke(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:372)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:903)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:698)
Caused by: java.lang.ClassCastException: android.preference.PreferenceCategory cannot be cast to android.preference.PreferenceScreen
       at org.wordpress.android.ui.prefs.SettingsFragment.hideNotificationBlogsCategory(SettingsFragment.java:181)
       at org.wordpress.android.ui.prefs.SettingsFragment.displayPreferences(SettingsFragment.java:252)
       at org.wordpress.android.ui.prefs.SettingsFragment.onCreate(SettingsFragment.java:126)
       at android.app.Fragment.performCreate(Fragment.java:2031)
       at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:863)
       at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1067)
       at android.app.BackStackRecord.run(BackStackRecord.java:834)
       at android.app.FragmentManagerImpl.execPendingActions(FragmentManager.java:1452)
       at android.app.Activity.performStart(Activity.java:6005)
       at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2288)
       at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2387)
       at android.app.ActivityThread.access$800(ActivityThread.java:151)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1303)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:135)
       at android.app.ActivityThread.main(ActivityThread.java:5254)
       at java.lang.reflect.Method.invoke(Method.java)
       at java.lang.reflect.Method.invoke(Method.java:372)
       at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:903)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:698)
```

--- a/WordPress/src/main/java/org/wordpress/android/ui/posts/PostUploadService.java
+++ b/WordPress/src/main/java/org/wordpress/android/ui/posts/PostUploadService.java
@@ -254,8 +254,8 @@ public class PostUploadService extends Service {
                 // add the tagline
                 SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(mContext);
 
-                if (prefs.getBoolean("wp_pref_signature_enabled", false)) {
-                    String tagline = prefs.getString("wp_pref_post_signature", "");
+                if (prefs.getBoolean(getString(R.string.pref_key_post_sig_enabled), false)) {
+                    String tagline = prefs.getString(getString(R.string.pref_key_post_sig), "");
                     if (!TextUtils.isEmpty(tagline)) {
                         String tag = "\n\n<span class=\"post_sig\">" + tagline + "</span>\n\n";
                         if (TextUtils.isEmpty(moreContent))

--- a/WordPress/src/main/java/org/wordpress/android/ui/prefs/NotificationSettingsFragment.java
+++ b/WordPress/src/main/java/org/wordpress/android/ui/prefs/NotificationSettingsFragment.java
@@ -57,7 +57,7 @@ public class NotificationSettingsFragment extends PreferenceFragment {
 
         // Add notifications group back in case it was previously removed from being logged out
         PreferenceCategory notificationTypesCategory = (PreferenceCategory) findPreference(
-                "wp_pref_notification_types");
+                getString(R.string.pref_notification_types));
         notificationTypesCategory.removeAll();
         SharedPreferences settings = PreferenceManager.getDefaultSharedPreferences(getActivity());
 
@@ -87,7 +87,8 @@ public class NotificationSettingsFragment extends PreferenceFragment {
                 }
             }
 
-            PreferenceCategory selectBlogsCategory = (PreferenceCategory) findPreference("wp_pref_notification_blogs");
+            PreferenceCategory selectBlogsCategory = (PreferenceCategory) findPreference(
+                    getActivity().getString(R.string.pref_notification_blogs));
             selectBlogsCategory.removeAll();
             for (int i = 0; i < mMutedBlogsList.size(); i++) {
                 StringMap<?> blogMap = mMutedBlogsList.get(i);
@@ -111,7 +112,8 @@ public class NotificationSettingsFragment extends PreferenceFragment {
             return;
         }
 
-        CheckBoxPreference notificationsEnabledCheckBox = (CheckBoxPreference) findPreference("wp_pref_notifications_enabled");
+        CheckBoxPreference notificationsEnabledCheckBox = (CheckBoxPreference) findPreference(
+                getString(R.string.pref_notifications_enabled));
         notificationsEnabledCheckBox.setOnPreferenceChangeListener(mNotificationsEnabledChangeListener);
     }
 
@@ -222,8 +224,8 @@ public class NotificationSettingsFragment extends PreferenceFragment {
             } else if (v.getId() == R.id.notifications8Hours) {
                 muteUntilMap.put("value", String.valueOf((System.currentTimeMillis() / 1000) + (3600 * 8)));
             }
-            CheckBoxPreference enabledCheckBoxPreference = (CheckBoxPreference) findPreference(
-                    "wp_pref_notifications_enabled");
+            CheckBoxPreference enabledCheckBoxPreference = (CheckBoxPreference) findPreference(getString(R.string
+                    .pref_notifications_enabled));
             enabledCheckBoxPreference.setChecked(false);
             mNotificationSettings.put("mute_until", muteUntilMap);
             mNotificationSettingsChanged = true;

--- a/WordPress/src/main/java/org/wordpress/android/ui/prefs/SettingsFragment.java
+++ b/WordPress/src/main/java/org/wordpress/android/ui/prefs/SettingsFragment.java
@@ -178,7 +178,7 @@ public class SettingsFragment extends PreferenceFragment {
     }
 
     private void hideNotificationBlogsCategory() {
-        PreferenceScreen preferenceScreen = (PreferenceScreen) findPreference(getActivity().getString(R.string.pref_key_notifications_section));
+        PreferenceScreen preferenceScreen = (PreferenceScreen) findPreference(getActivity().getString(R.string.pref_key_settings_root));
         PreferenceCategory blogs = (PreferenceCategory) findPreference("wp_pref_notification_blogs");
         if (preferenceScreen != null && blogs != null) {
             preferenceScreen.removePreference(blogs);
@@ -246,7 +246,6 @@ public class SettingsFragment extends PreferenceFragment {
     }
 
     void displayPreferences() {
-        // Post signature
         if (WordPress.wpDB.getNumVisibleAccounts() == 0) {
             hidePostSignatureCategory();
             hideNotificationBlogsCategory();

--- a/WordPress/src/main/java/org/wordpress/android/ui/prefs/SettingsFragment.java
+++ b/WordPress/src/main/java/org/wordpress/android/ui/prefs/SettingsFragment.java
@@ -178,8 +178,10 @@ public class SettingsFragment extends PreferenceFragment {
     }
 
     private void hideNotificationBlogsCategory() {
-        PreferenceScreen preferenceScreen = (PreferenceScreen) findPreference(getActivity().getString(R.string.pref_key_settings_root));
-        PreferenceCategory blogs = (PreferenceCategory) findPreference("wp_pref_notification_blogs");
+        PreferenceScreen preferenceScreen =
+                (PreferenceScreen) findPreference(getActivity().getString(R.string.pref_key_settings_root));
+        PreferenceCategory blogs =
+                (PreferenceCategory) findPreference(getActivity().getString(R.string.pref_notification_blogs));
         if (preferenceScreen != null && blogs != null) {
             preferenceScreen.removePreference(blogs);
         }

--- a/WordPress/src/main/res/values/strings.xml
+++ b/WordPress/src/main/res/values/strings.xml
@@ -357,7 +357,7 @@
     <string name="search_location">Search</string>
     <string name="edit_location">Edit</string>
 
-    <!-- preferences -->
+    <!-- Preference keys -->
     <string name="pref_key_settings_root" translatable="false">wp_pref_root</string>
     <string name="pref_key_wpcom" translatable="false">wp_pref_wpcom</string>
     <string name="pref_key_self_hosted" translatable="false">wp_pref_self_hosted_blogs</string>
@@ -379,6 +379,11 @@
     <string name="pref_key_help_and_support" translatable="false">wp_pref_help_and_support</string>
     <string name="pref_key_signout_section" translatable="false">wp_pref_app_signout_section</string>
     <string name="pref_key_signout" translatable="false">wp_pref_sign_out</string>
+    <string name="pref_notification_blogs" translatable="false">wp_pref_notification_blogs</string>
+    <string name="pref_notifications_enabled" translatable="false">wp_pref_notifications_enabled</string>
+    <string name="pref_notification_types" translatable="false">wp_pref_notification_types</string>
+
+    <!-- preferences -->
     <string name="help_and_support">Help and support</string>
     <string name="open_source_licenses">Open source licenses</string>
     <string name="wpcom_blogs">WordPress.com blogs</string>

--- a/WordPress/src/main/res/xml/notification_settings.xml
+++ b/WordPress/src/main/res/xml/notification_settings.xml
@@ -3,15 +3,15 @@
                   android:key="wp_pref_notifications_root">
     <CheckBoxPreference
         android:defaultValue="true"
-        android:key="wp_pref_notifications_enabled"
+        android:key="@string/pref_notifications_enabled"
         android:title="@string/enable_notifications"/>
     <PreferenceCategory
         android:title="@string/types"
-        android:key="wp_pref_notification_types">
+        android:key="@string/pref_notification_types">
     </PreferenceCategory>
     <PreferenceCategory
         android:title="@string/blogs"
-        android:key="wp_pref_notification_blogs">
+        android:key="@string/pref_notification_blogs">
     </PreferenceCategory>
     <PreferenceCategory android:title="@string/options">
         <CheckBoxPreference

